lib/hooksRegistry.js

```diff
@@ -98,6 +98,11 @@ async function registerConfigHooks(hooks, templateDir, templateConfig) {
  * @param {Array} config List of hooks configured in template configuration
  */
 function addHook(hooks, mod, config) {
+  // for ESM `export default {}`
```

Comment:
what does it mean, what esm has to do with this part?

Reply:
For example. If you write export for hooks in CommonJS, then you have:

```js
module.exports = {
  'generate:before': (generator) => {
  }
};
```

But if you are using ESM in TS, you can write:

```ts
export default {
  'generate:before': (generator) => {
  }
}
```

Unfortunately, then it is transpiled to:

```js
module.exports = {
  default: {
      'generate:before': (generator) => {
      }
  }
};
```

Reply:
oh, ok, you just made a comment to explain the `if`. Can you make it more descriptive 🙏🏼 I totally didn't map it in my head, probably because I use ESM very rarely. So please write this comment for people like me too 😄